controllers/servers/host_definer/watcher/watcher_helper.py

```diff
@@ -0,0 +1,469 @@
+import base64
+import datetime
+import os
+from kubernetes import client, config, dynamic
+from kubernetes.client import api_client
+from kubernetes.client.rest import ApiException
+
+from controllers.common.csi_logger import get_stdout_logger
+from controllers.servers.host_definer.common import settings
+from controllers.servers.host_definer.common.types import DefineHostRequest
+from controllers.servers.host_definer.storage_manager.host import StorageHostManager
+import controllers.servers.host_definer.watcher.exceptions as exceptions
+
+SECRET_IDS = {}
+NODES = {}
+logger = get_stdout_logger()
+
+
+class WatcherHelper:
+    def __init__(self):
+        self.storage_host_manager = StorageHostManager()
+        self._load_cluster_configuration()
+        self.dynamic_client = self._get_dynamic_client()
+        self.storage_api = client.StorageV1Api()
+        self.core_api = client.CoreV1Api()
+        self.custom_object_api = client.CustomObjectsApi()
+        self.csi_nodes_api = self._get_csi_nodes_api()
+        self.host_definitions_api = self._get_host_definitions_api()
+
+    def _get_dynamic_client(self):
+        return dynamic.DynamicClient(api_client.ApiClient(
+            configuration=self._load_cluster_configuration()))
+
+    def _load_cluster_configuration(self):
+        return config.load_incluster_config()
+
+    def _get_csi_nodes_api(self):
+        return self.dynamic_client.resources.get(
+            api_version=settings.STORAGE_API_VERSION,
+            kind=settings.CSINODE_KIND)
+
+    def _get_host_definitions_api(self):
+        return self.dynamic_client.resources.get(
+            api_version=settings.CSI_IBM_BLOCK_API_VERSION,
+            kind=settings.HOSTDEFINITION_KIND)
+
+    def verify_nodes_defined(self, host_request):
+        for node in NODES:
+            host_request.node_id = self.get_node_id_from_node_name(node)
+            self._verify_host_defined_and_has_host_definition(host_request)
+
+    def _verify_host_defined_and_has_host_definition(self, host_request):
+        if self._is_host_already_defined(host_request):
+            return
+        response = self.verify_host_defined_on_storage_and_on_cluster(host_request)
+        if response.error_message:
+            self._verify_host_definition_in_phase(host_request, settings.PENDING_CREATION_PHASE)
+            self._create_event_to_host_definition_from_host_request(
+                host_request, response.error_message)
+
+    def _is_host_already_defined(self, host_request):
```

Comment:
please consider:
```
def _is_host_defined(self, host_request):
     host_definition_name = self.get_host_definition_name(host_request)
     return self._is_host_definition_in_ready_state(host_definition_name)
```
(`get_host_definition_name` getting the `host_request` so it can ask for `get_node_name_from_node_id` by itself)
logger will be like [here](https://github.com/IBM/ibm-block-csi-driver/pull/514/files#r908741892)